File: tor_selden-csharp/tor_selden-csharp/Day11.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace tor_selden_csharp
{
    public class Point
    {
        int x;
        int y;
        int z;

        public Point(int x, int y, int z)
        {
            this.x = x;
            this.y = y;
            this.z = z;
        }

        public int Distance(Point b)
        {
            return (Math.Abs(x - b.x) + Math.Abs(y - b.y) + Math.Abs(z - b.z)) / 2;            
        }
    }

    class Day11
    {
        static string input = File.ReadAllText(Path.Combine(Program.BasePath, "input11.txt"));

        public static void AB()
        {
            var directions = input.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries).ToList();

            int x = 0;
            int y = 0;
            int z = 0;
            int maxDist = 0;

            var p1 = new Point(x, y, z);

            foreach (var direction in directions)
            {
                switch (direction)
                {
                    case "n":
                        y++;
                        z--;
                        break;
                    case "ne":
                        x++;
                        z--;
                        break;
                    case "se":
                        ++x;
                        --y;
                        break;
                    case "s":
                        y--;
                        z++;
                        break;
                    case "sw":
                        x--;
                        z++;
                        break;
                    case "nw":
                        y++;
                        x--;
                        break;
                    default:
                        break;
                }

                var temp = new Point(x, y, z);
                maxDist = p1.Distance(temp) > maxDist ? p1.Distance(temp) : maxDist;

            }

            var p2 = new Point(x, y, z);

            var dist = p1.Distance(p2);

            Console.WriteLine(dist); //A
            Console.WriteLine(maxDist); //B
        }        
    }
}

File: plilja-python3/day11/day11.py
from collections import defaultdict

def step1(directions):
    return run(directions)[0]


def step2(directions):
    return run(directions)[1]


def run(directions):
    def dist(x, y):
        if (x < 0 and y < 0) or (x > 0 and y > 0):
            return max(abs(x), abs(y))
        else:
            return abs(x) + abs(y)

    x, y = 0, 0
    max_dist = -1
    for dr in directions:
        if dr == 'n':
            dx, dy = -1, -1
        elif dr == 'ne':
            dx, dy = 0, -1
        elif dr == 'se':
            dx, dy = 1, 0
        elif dr == 's':
            dx, dy = 1, 1
        elif dr == 'sw':
            dx, dy = 0, 1
        elif dr == 'nw':
            dx, dy = -1, 0
        x += dx
        y += dy
        max_dist = max(max_dist, dist(x, y))

    return dist(x, y), max_dist


directions = input().split(',')
print(step1(directions))
print(step2(directions))
